pkg/context.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Chair of Applied Cryptography, Technische Universität
+// Darmstadt, Germany. All rights reserved. This file is part of
+// perun-eth-demo. Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package pkg
+
+import (
+	"context"
+)
+
+// Context wraps a golang context/Context with a cancel function when available.
+// See https://golang.org/pkg/context/#Context
+type Context struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// Background returns the Background context that is never cancelled.
+// ref https://golang.org/pkg/context/#Background
+func Background() *Context {
```

Comment:
Since you're not in a specific `context` package, this name is not clear. Better rename to
```suggestion
func ContextBackground() *Context {
```

Reply:
Ok